src/nvim/api/private/helpers.c

```diff
@@ -404,9 +404,18 @@ void set_option_to(void *to, int type, String name, Object value, Error *err)
       return;
     }
 
-    set_option_value_for(name.data, 0, value.data.string.data,
-            opt_flags, type, to, err);
+    stringval = (char *)value.data.string.data;
   }
+
+  const scid_T save_current_SID = current_SID;
+  current_SID = channel_id == LUA_INTERNAL_CALL ? SID_LUA : SID_API_CLIENT;
```

Comment:
I guess `is_internal_call()` would not make sense here.

In the future I think we should somehow track the Lua function name or minimal stacktrace.